src/plugins/data/public/ui/dataset_selector/index.tsx

```diff
@@ -34,6 +35,7 @@ const ConnectedDatasetSelector = ({ onSubmit }: ConnectedDatasetSelectorProps) =
       const query = queryString.getInitialQueryByDataset(dataset);
       queryString.setQuery(query);
       onSubmit!(queryString.getQuery());
+      services.data.query.queryString.getDatasetService().addRecentDataset(dataset);
```

Comment:
```typescript
queryString.getDatasetService().addRecentDataset(initialDataset);
```

Reply:
@ashwin-pc do you think we should be just handling this is the query string manager? like updating the recents based on the query?

Reply:
Yes that would make it less prone to issues like when datasets are selected using ways that arent the dataset selector. Wouldnt block on that being a requirement, but that is a cleaner approach

Reply:
@sejli maybe could be an open issue